tests/mlmodel_sklearn/test_linear_models.py

```diff
@@ -0,0 +1,342 @@
+            ("Function/MLModel/Sklearn/Named/PassiveAggressiveRegressor.predict", 2),
+            ("Function/MLModel/Sklearn/Named/PassiveAggressiveRegressor.score", 1),
+        ],
+        "Perceptron": [
+            ("Function/MLModel/Sklearn/Named/Perceptron.fit", 1),
+            ("Function/MLModel/Sklearn/Named/Perceptron.predict", 2),
+            ("Function/MLModel/Sklearn/Named/Perceptron.score", 1),
+        ],
+        "Ridge": [
+            ("Function/MLModel/Sklearn/Named/Ridge.fit", 1),
+            ("Function/MLModel/Sklearn/Named/Ridge.predict", 2),
+            ("Function/MLModel/Sklearn/Named/Ridge.score", 1),
+        ],
+        "RidgeCV": [
+            ("Function/MLModel/Sklearn/Named/RidgeCV.fit", 1),
+            ("Function/MLModel/Sklearn/Named/RidgeCV.predict", 2),
+            ("Function/MLModel/Sklearn/Named/RidgeCV.score", 1),
+        ],
+        "RidgeClassifier": [
+            ("Function/MLModel/Sklearn/Named/RidgeClassifier.fit", 1),
+            ("Function/MLModel/Sklearn/Named/RidgeClassifier.predict", 2),
+            ("Function/MLModel/Sklearn/Named/RidgeClassifier.score", 1),
+        ],
+        "RidgeClassifierCV": [
+            ("Function/MLModel/Sklearn/Named/RidgeClassifierCV.fit", 1),
+            ("Function/MLModel/Sklearn/Named/RidgeClassifierCV.predict", 2),
+            ("Function/MLModel/Sklearn/Named/RidgeClassifierCV.score", 1),
+        ],
+        "TheilSenRegressor": [
+            ("Function/MLModel/Sklearn/Named/TheilSenRegressor.fit", 1),
+            ("Function/MLModel/Sklearn/Named/TheilSenRegressor.predict", 2),
+            ("Function/MLModel/Sklearn/Named/TheilSenRegressor.score", 1),
+        ],
+        "RANSACRegressor": [
+            ("Function/MLModel/Sklearn/Named/RANSACRegressor.fit", 1),
+            ("Function/MLModel/Sklearn/Named/RANSACRegressor.predict", 1),
+            ("Function/MLModel/Sklearn/Named/RANSACRegressor.score", 1),
+        ],
+    }
+    expected_transaction_name = "test_linear_models:_test"
+    if six.PY3:
+        expected_transaction_name = (
+            "test_linear_models:test_below_v1_1_model_methods_wrapped_in_function_trace.<locals>._test"
+        )
+
+    @validate_transaction_metrics(
+        expected_transaction_name,
+        scoped_metrics=expected_scoped_metrics[linear_model_name],
+        rollup_metrics=expected_scoped_metrics[linear_model_name],
+        background_task=True,
+    )
+    @background_task()
+    def _test():
+        run_linear_model(linear_model_name)
+
+    _test()
+
+
+@pytest.mark.skipif(SKLEARN_VERSION < (1, 1, 0), reason="Requires sklearn >= v1.1")
+@pytest.mark.parametrize(
+    "linear_model_name",
+    [
+        "PoissonRegressor",
+        "GammaRegressor",
+        "TweedieRegressor",
+        "QuantileRegressor",
+        "SGDClassifier",
+        "SGDRegressor",
+        "SGDOneClassSVM",
+    ],
+)
+def test_above_v1_1_model_methods_wrapped_in_function_trace(linear_model_name, run_linear_model):
```

Comment:
It looks like there are more than just these models above 1.1.0. We should test everything we can on each version since the models are sometimes in different files and we need to make sure we instrument those files.

Reply:
Sorry if I missed something, but I could not find any more models that were missed above 1.1 (looking in the latest release on main) unless you were referring to the ones in sgd_fast, which are Cython files.

Reply:
I don't see RidgeClassifier sklearn/linear_model/_ridge.py mentioned in this test but I do see it on main for example. I see a lot of the model classes in test_below_v1_1 on main. OrthogonalMatchingPursuitCV in sklearn/linear_model/_omp.py:class is another one. I think we just need to copy the majority of those models in below 1.1 into this above 1.1 as well.

Reply:
Ooh, I see what you mean.  The ones in the "below_ v1_1" test do not have a skipif condition, so those models are in all versions, but the ones in the "above_v1_1" do, so those only test the ones added after v1.1.  Maybe I should change the test name to be less confusing?

Reply:
Ah I see. I was being really dumb about that. I think the name is fine as is.